### ISSUE TITLE
Feature Request: structured output - versioning/tracing for nested pydantic models

### ISSUE THREAD

**Issue by AxelBL** [MadcowD/ell#300]
I use nested pydantic models in the structured output. ell only keeps track of the first level pydantic model in versioning and tracing.
Example:

class Sub(BaseModel):
    fields...

class Main(BaseModel):
   ...
   Subs: List[Sub]
   ...

@ell.complex(model="gpt-4o-mini", response_format=Main)
def ...

=> the ell.db will only keep track of and show Main in ell-studio, but not any nested classes like Sub

Please look for nested classes and add them to tracing/versioning. I.e., detect changes in nested classes like Sub and version accordingly.

Thanks a lot!

**Comment by MadcowD** [comment]
Yes this is a great idea, we need to recurse the lexical closure on types. this should come in a month or so i hope thats okay.

**Comment by MadcowD** [comment]
I can prioritize this if it's really needed :) Whats your use case?
